src/parser.y

```diff
@@ -225,7 +225,7 @@ static block constant_fold(block a, block b, int op) {
     int cmp = jv_cmp(jv_a, jv_b);
 
     switch (op) {
-    case '+': res = jv_number(na + nb); break;
+    case '+': res = jv_number_add(jv_copy(jv_a), jv_copy(jv_b)); break;
```

Comment:
Isn't it too late to copy the arguments here? jv_cmp above has already consumed those, hasn't it?

Reply:
You're right, it has. I'll fix that.